azure_arc_k8s_jumpstart/cluster_api/capi_keyvault_extension/capi_keyvault_k8s_extension.sh

```diff
@@ -0,0 +1,132 @@
+
+echo "Checking if you have up-to-date Azure Arc Az CLI 'connectedk8s' extension..."
+az extension show --name "connectedk8s" &> extension_output
+if cat extension_output | grep -q "not installed"; then
+az extension add --name "connectedk8s"
+rm extension_output
+else
+az extension update --name "connectedk8s"
+rm extension_output
+fi
+echo ""
+
+echo "Checking if you have up-to-date Azure Arc Az CLI 'k8s-extension' extension..."
+az extension show --name "k8s-extension" &> extension_output
+if cat extension_output | grep -q "not installed"; then
+az extension add --name "k8s-extension"
+rm extension_output
+else
+az extension update --name "k8s-extension"
+rm extension_output
+fi
+echo ""
+
+echo "Creating Key Vault"
+az group create --name $keyVaultResourceGroup --location $keyVaultLocation
+az keyvault create --name $keyVaultName --resource-group $keyVaultResourceGroup --location $keyVaultLocation
+
+echo "Creating Key Vault Secret"
+az keyvault secret set --vault-name $keyVaultName --name dbusername --value "HelloArc!"
+
+echo "Create Azure Key Vault Kubernetes extension instance"
+az k8s-extension create --name $k8sExtensionName --extension-type Microsoft.AzureKeyVaultSecretsProvider --scope cluster --cluster-name $arcClusterName --resource-group $resourceGroup --cluster-type connectedClusters --release-train preview --release-namespace kube-system --configuration-settings 'secrets-store-csi-driver.enableSecretRotation=true' 'secrets-store-csi-driver.syncSecret.enabled=true'
+
+# Create a namespace for your ingress resources
+kubectl create ns $namespace
+
+# Create the Kubernetes secret with the service principal credentials
+kubectl create secret generic secrets-store-creds --namespace $namespace --from-literal clientid=${appId} --from-literal clientsecret=${password}
+kubectl --namespace $namespace label secret secrets-store-creds secrets-store.csi.k8s.io/used=true
+
+# Deploy SecretProviderClass
+echo "Creating Secret Provider Class"
+cat <<EOF | kubectl apply -n $namespace -f -
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: azure-kv-sync
+spec:
+  provider: azure
+  secretObjects:   
+    - secretName: dbusername
+      type: Opaque
+      data:
+        - objectName: dbusername
+          key: username
+  parameters:
+    usePodIdentity: "false"
+    useVMManagedIdentity: "false"
+    userAssignedIdentityID: ""
+    keyvaultName: "${keyVaultName}"
+    objects: |
+      array:
+        - |
+          objectName: dbusername             
+          objectType: secret
+          objectVersion: ""
+    tenantId: "${tenantId}"
+EOF
+
+# Create the pod with volume referencing the secrets-store.csi.k8s.io driver
+echo "Deploying App referencing the secret"
+cat <<EOF | kubectl apply -n $namespace -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-secrets-sync
+spec:
+  containers:
+  - name: busybox
+    image: k8s.gcr.io/e2e-test-images/busybox:1.29-1
```

Comment:
This image tag is also valid but could we use `k8s.gcr.io/e2e-test-images/busybox:1.29` instead? That's the one we're using upstream

Reply:
Updated the image tag